pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/segmentgenerationandpush/SegmentGenerationAndPushTaskGenerator.java

```diff
@@ -177,6 +180,76 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
     return pinotTaskConfigs;
   }
 
+  @Override
+  public List<PinotTaskConfig> generateAdhocTasks(TableConfig tableConfig, Map<String, String> taskConfigs)
+      throws Exception {
+    String taskUUID = UUID.randomUUID().toString();
+    // Only generate tasks for OFFLINE tables
+    String offlineTableName = tableConfig.getTableName();
+    if (tableConfig.getTableType() != TableType.OFFLINE) {
+      LOGGER.warn("Skip generating SegmentGenerationAndPushTask for non-OFFLINE table: {}", offlineTableName);
+      return ImmutableList.of();
```

Comment:
Should we not throw an exception to return this error to the caller of the `/tasks/execute` API?

Reply:
this is not an exception, the external caller will call this twice for realtime/offline table separately.

Reply:
I see your point,  you mean if users call the REALTIME table, then it should be a 4xx response, and if users call the hybrid table name, then it will be an empty 2xx response right?

Reply:
right

Reply:
I see. Then I think the task gen call can return null, and the external will skip pushing to the map. And eventually, this call will throw an exception if the response map is empty(which means no task gen runs successfully for any table)

Reply:
The outer call will handle the empty list by not putting it into the results mapping